Review link command description for Managed C (#1346)
MicroEJ/docs: ApplicationDeveloperGuide/managedc/compilation.rst

Once the C files are compiled into object files (``.o`` files),
the next step is to link them together to produce a Wasm module (``.wasm`` file).
Use ``clang`` as the linker driver to generate the final ``.wasm`` module. 
It automatically calls ``wasm-ld``, the Wasm linker.

> The following options are recommended to ensure the produced ``.wasm`` binary is compatible with the MicroEJ VEE:

Author: ```suggestion
The following options are recommended to ensure the produced ``.wasm`` binary is compatible with MICROEJ VEE:
```

> * ``-Wl,--allow-undefined``: Allows undefined symbols, which is necessary when the Wasm module references extern symbols defined in Java.

Author: ```suggestion
* ``-Wl,--allow-undefined``: Allows undefined symbols, which is necessary when the Wasm module references extern symbols defined by the Java host.
```

* ``-Wl,--stack-first``: Places the stack at the begining of the linear memory. Refer to :ref:`managedc.linear.memory.layout` for more information.
* ``-z stack-size=n``: Adjusts the stack size to 'n' bytes. Consult :ref:`managedc.linear.memory.size.stack` for further information.

For example, if you have the object file ``src/main/resources/my_app.o`` that defines a ``main`` entry point,
the following command will generate a Wasm module including this entry point along with all transitive dependencies from the :ref:`WASI Libc <wasm.wasi>`.
Any unresolved symbols in the object file are expected to be provided by the Java host.

    .. code:: console

        [path_to_wasi_sdk]/bin/clang -Wl,--stack-first -z stack-size=4096 -Wl,--allow-undefined -o src/main/resources/my_app.wasm src/main/resources/my_app.o

When your object file defines a ``main`` function, the linker automatically exports a function named ``_start``. 
This function performs initialization tasks before invoking the user-defined ``main``. 
You must bind the ``_start`` function — not ``main`` — to a Java method.

> Additionally, any other functions that should be callable from the Java application must be explicitly exported using the ``-Wl,--export=[function_name]`` linker option (this does not apply to ``main``, which is invoked internally by ``_start``).  

Author: ```suggestion
Additionally, any other functions that should be callable from the Java Application must be explicitly exported using the ``-Wl,--export=[function_name]`` linker option (this does not apply to ``main``, which is invoked internally by ``_start``).  
```

    You can then reference this file in the command line using the ``-Wl,@exports.txt`` option. 

If your module is composed of multiple C files, make sure to append all the object files to the command line.
The resulting file, ``my_app.wasm``, will be a linked Wasm module containing all functions, globals, and constants defined across those object files.

This Wasm module can then be :ref:`bound to a Java class <managedc.bind.module>` for integration with your application.
For additional details and alternative configurations, see the :ref:`managedc.link.command_line_options` section.

.. _managedc.link.command_line_options:

Linker Options
``````````````

> Here are other linker options you might need:

Author: There are many references to this section that assume all declared options.
If we shrink the defined options, so I think we definitely should fully remove this section. I will suggest an other PR.

* `-Wl,--export=[function_name]` already declared above
* `-nostdlib` is already defined in <managedc.link.nostdlib> => we have to add `-Wl,--no-entry`
* `--initial-memory` is defined in the linear memory page
